### PR TITLE
Added support for keyquest keyboards

### PR DESCRIPTION
**Pull request by keyquesttech**
Added support for keyquest keyboards and "Enclave-1"

## Description

I've added the "keyquest" folder which contains the enclave files to build firmware for Enclave-1

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keyquest/enclave/config.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#pragma once
```
Comment (drashna) [inline keyboards/keyquest/enclave/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/keyquest/enclave/config.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#define MATRIX_COL_PINS { B4, B7, C7 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+/* RGB light setup*/
+#define RGB_DI_PIN B5
+#define RGBLED_NUM 2
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/keyquest/enclave/config.h:24]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/keyquest/enclave/info.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+    "manufacturer": "keyquest",
+    "keyboard_name": "Enclave-1",
+    "maintainer": "the keyquest team",
+    "layouts": {
+        "LAYOUT_macro_3x3": {
```
Comment (drashna) [inline keyboards/keyquest/enclave/info.json:6]: This should probably just be `LAYOUT`.  Eg
```suggestion
        "LAYOUT": {
```

Though, changing it here would require changing it anywhere else it's being used, too

File: keyboards/keyquest/enclave/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/keyquest/enclave/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/keyquest/enclave/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        break;
+
+    case WINCLIP:
+        if (record->event.pressed) {
+           SEND_STRING(SS_LGUI("v"));
+        }
+        break;
+    }
+    return true;
+};
```
Comment (drashna) [inline keyboards/keyquest/enclave/keymaps/default/keymap.c:33]: No need for any of this.  

`C(KC_C)`, `C(KC_V)` and `G(KC_V)` can be used in the keymap directly, without the need for custom macros.

File: keyboards/keyquest/enclave/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/keyquest/enclave/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/keyquest/enclave/readme.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Enclave-1
+
+![ok](https://user-images.githubusercontent.com/109129639/178805184-3935bcfa-2cb5-438d-8172-406f9a6d3bbd.png)
```
Comment (drashna) [inline keyboards/keyquest/enclave/readme.md:3]: Please rehost this image on Imgur.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keyquest/enclave/keymaps/default/keymap.c (hunk 5 above)
Comment (keyquesttech) [reply keyboards/keyquest/enclave/keymaps/default/keymap.c:46]: This was done intentionally to have a template to move onto more complex macros. If it needs to be changed i will update but this is the ideal default keymap

**Review by drashna — APPROVED** [review]

File: keyboards/keyquest/enclave/readme.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Enclave-1
+
+![ok](https://i.imgur.com/67HZEJB.jpg)
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/readme.md:3]:
```suggestion
![ok](https://i.imgur.com/67HZEJBh.jpg)
```

File: keyboards/keyquest/enclave/readme.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+Enclave 1 is a fully programmable 9-key macropad + usb hub ideal for those constantly on the move that need to adapt to any situation. Besides being a full-fledged macropad, it can also be used as the main hub for all your other devices such as keyboards, dongles, drives, etc.
+
+* Keyboard Maintainer: [KeyQuest](https://github.com/KeyQuest)
+
+Compile example for this keyboard (after setting up your build environment):
+
+    qmk compile -kb keyquest/enclave -km default
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/readme.md:11]:
```suggestion
    make keyquest/enclave:default
```

File: keyboards/keyquest/enclave/rules.mk [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+# Change yes to no to disable the feature
+
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/rules.mk:17]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Mathiaszmrga** [comment]
It should be good to go

File: keyboards/keyquest/enclave/rules.mk (hunk 10 above)
Comment (keyquesttech) [reply keyboards/keyquest/enclave/rules.mk]: All changes done and verified

File: keyboards/keyquest/enclave/rules.mk [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+# Change yes to no to disable
+
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/rules.mk:17]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keyquest/enclave/rules.mk (hunk 11 above)
Comment (keyquesttech) [reply keyboards/keyquest/enclave/rules.mk]: fixed on last commit. such a small thing idk how i missed it, sorry for the mistake this time it should all be fine

**Comment by keyquesttech** [comment]
Sorry for the ping when it wasn't ready. im 99% this time it is ready.

File: keyboards/keyquest/enclave/rules.mk [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+# Change yes to no to disable
+
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underlight.
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/rules.mk:17]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/keyquest/enclave/rules.mk [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+# Change yes to no to disable
+
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/rules.mk:9]:
```suggestion
#
```

File: keyboards/keyquest/enclave/info.json [hunk 14]
```diff
@@ -0,0 +1,22 @@
+{
+    "manufacturer": "keyquest",
+    "keyboard_name": "Enclave-1",
+    "maintainer": "the keyquest team",
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/info.json:4]:
```suggestion
    "maintainer": "the keyquest team",
    "usb": {
        "vid": "0x1117",
        "pid": "0x0E0E",
        "device_version": "0.0.1"
    },
```

File: keyboards/keyquest/enclave/config.h [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+*/
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x1117
+#define PRODUCT_ID      0x0E0E
+#define DEVICE_VER      0x0001
+#define PRODUCT         Enclave-1
+
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/config.h:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by keyquesttech** [comment]
Hopefully its ok now 🙄

File: keyboards/keyquest/enclave/rules.mk [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+# Change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow.
```
Comment (fauxpark) [inline keyboards/keyquest/enclave/rules.mk:17]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by keyquesttech** [comment]
Fixed issue with the period. Sorry for missing that 😓

**Review by drashna — APPROVED** [review]

**Comment by keyquesttech** [comment]
Hello @fauxpark, could you check the changes, i'm confident this time it's fine. Thanks for your time.

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**